code/modules/projectiles/guns/energy/laser.dm

```diff
@@ -304,6 +304,24 @@
 	charge_cost = 1500 //You got 1 shot...
 	projectile_type = /obj/projectile/beam/heavylaser //But it hurts a lot
 	cell_type = /obj/item/cell/device/weapon
+	unstable = 1
+
+/obj/item/gun/energy/zip/consume_next_projectile(mob/user as mob)
+	. = ..()
+	if(.)
+		if(unstable)
+			switch(rand(1,100))
```

Comment:
you can just do a if(prob(10)) if you want to run with a 10% chance

Reply:
> you can just do a if(prob(10)) if you want to run with a 10% chance

Oh awesome! I'll have to get with you later to go over that.